src/Components/DrillIllustration.js

```diff
@@ -260,14 +203,132 @@ const DrillIllustration = props => {
+        <View style={styles.line}>
+          <View>
+            {!isFirstStep && (
+              <TouchableOpacity
+                onPress={() => {
+                  carouselRef.current.snapToPrev();
+                }}
+              >
+                <MaterialCommunityIcons name="chevron-double-left" color={theme.COLOR_PRIMARY} size={26} />
+              </TouchableOpacity>
+            )}
+          </View>
+          <Text style={styles.title}>{title}</Text>
+          <View>
+            {!isLastStep && (
+              <TouchableOpacity
+                onPress={() => {
+                  carouselRef.current.snapToNext();
+                }}
+              >
+                <MaterialCommunityIcons name="chevron-double-right" color={theme.COLOR_PRIMARY} size={26} />
+              </TouchableOpacity>
+            )}
+          </View>
+        </View>
+        <View style={styles.animation}>
+          <Animation widthRatio={1} heightRatio={props.minimal ? 2 / 5 : 1 / 2} animation={illustrationSource} />
+        </View>
+      </>
+    );
+  };
+
+  const renderStep = ({ item, index }) => {
+    if (!currentStep) {
+      return <View />; // bad state, but let's not crash
+    } else {
+      switch (item.illustrationType) {
+        case IllustrationType.ANIMATION:
+          return displayAnimation(item, index);
+        case IllustrationType.YOUTUBE:
+          return displayYoutube(item);
+        case IllustrationType.VIMEO:
+          return displayVimeo(item);
+        default:
+          return <Text>No visual content for this drill</Text>;
+      }
+    }
+  };
+
+  const pagination = () => {
+    return (
+      <Pagination
+        dotsLength={props.drill.steps.length}
+        activeDotIndex={activeIndex}
+        containerStyle={{
+          backgroundColor: theme.BACKGROUND_COLOR_LIGHT,
+          paddingVertical: 0,
+        }}
+        dotStyle={{
+          width: 8,
+          height: 8,
+          borderRadius: 5,
+          marginHorizontal: 5,
+          backgroundColor: theme.GRADIENT_FIRST_COLOR,
+        }}
+        inactiveDotStyle={{
+          backgroundColor: theme.COLOR_SECONDARY,
+        }}
+        inactiveDotOpacity={0.4}
+        inactiveDotScale={0.6}
+      />
+    );
+  };
+
+  const drillTypes = () => {
+    if (props.drill.type === 'frisbee') {
+      return (
+        <SafeAreaView style={styles.container}>
+          <Carousel
+            layout="default"
+            ref={carouselRef}
+            data={props.drill.steps}
+            sliderWidth={screenDimension.width}
+            itemWidth={screenDimension.width}
+            renderItem={renderStep}
+            onSnapToItem={index => setActiveIndex(index)}
+          />
+          <View style={styles.pagination}>{pagination()}</View>
+        </SafeAreaView>
+      );
+    } else {
+      return (
+        <GestureRecognizer style={styles.container} onSwipeLeft={checkAnimationFast} config={swipeConfig}>
+          {checkSwitch()}
```

Comment:
This if/else statement means that frisbee drills are ALWAYS going to have animation (displayed within a carousel)
And fitness drils may have different illustrations : animation (not in a carousel), youtube and vimeo

Are we sure about this?
If yes, I think we should split this component and have 1 component for FrisbeeDrillIllustration and another for FitnessDrillIllustration, because the display is now different. That would make it easier to understand what's happening

Reply:
Yes I am not 100% sure about this. I think if in the future there is a drill explain with a video and a animation we should use something else than the carousel. The problem now is, if we are explaining a drill with 2 videos. they both start when we arrive on the drillpage.

Reply:
After reflexion, a frisbee drill will always use a carousel. If there is video inside ,these videos have to on "pause" at the begining

Reply:
Yep, I agree, when we add controls on video, the user will be able to play them when he's ready